GOT-StudentVersion/GOTViewControllers/ViewController.swift

```diff
@@ -0,0 +1,118 @@
+import UIKit
+
+class ViewController: UIViewController,UITableViewDataSource,UITableViewDelegate {
+    //MARK: Outlets & Actions
+    @IBOutlet weak var searchBarOutlet: UISearchBar!
+    @IBOutlet weak var segmentedControl: UISegmentedControl!
+     @IBOutlet weak var tableView: UITableView!
+    @IBAction func segmentedControlToggled(_ sender: UISegmentedControl) {
+        // reloads tableview when  segmentedControll is toggled
+        tableView.reloadData()
+    }
+    //MARK: Variables
+    var isSearching = false
+    var allGOTSeasons = [[GOTEpisode]](){
+        didSet {
+            tableView.reloadData()
+        }
+    }
+    var filteredEpisodes = [[GOTEpisode]]() { // This will hold the search results
+        didSet {
+            tableView.reloadData()
+        }
+    }
+    
+   
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        setupViewDidLoad()
+      segmentedControl.selectedSegmentIndex = 0
+       
+        // giving data to allGOTSeasons when the view loads
+        allGOTSeasons = GOTEpisode.createGOTDataMatrix() //
+    }
+    //MARK: Functions
+    
+    func setupViewDidLoad() {
+        tableView.dataSource = self
+        tableView.delegate = self
+        searchBarOutlet.delegate = self
+    }
+    
+    
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int{
+        return isSearching ? filteredEpisodes[section].count : allGOTSeasons[section].count
+    }
+    func tableView(_ tableView: UITableView, titleForHeaderInSection section: Int) -> String? {
+        
+        let titleHeader = isSearching ? filteredEpisodes[section][0].season : allGOTSeasons[section][section].season
+        return "Season \(titleHeader)"
+    }
+    func numberOfSections(in tableView: UITableView) -> Int {
+        
+        return isSearching ? filteredEpisodes.count :  allGOTSeasons.count
```

Comment:
I love this ternary operator, makes things succinct and clear for if they are searching or not